task/backend/coordinator/coordinator.go

```diff
@@ -151,3 +151,15 @@ func (c *Coordinator) DeleteOrg(ctx context.Context, orgID platform.ID) error {
 func (c *Coordinator) CancelRun(ctx context.Context, taskID, runID platform.ID) error {
 	return c.sch.CancelRun(ctx, taskID, runID)
 }
+
+func (c *Coordinator) ManuallyRunTimeRange(ctx context.Context, taskID platform.ID, start, end, requestedAt int64) (*backend.StoreTaskMetaManualRun, error) {
+	r, err := c.Store.ManuallyRunTimeRange(ctx, taskID, start, end, requestedAt)
+	if err != nil {
+		return r, err
+	}
+	t, m, err := c.Store.FindTaskByIDWithMeta(ctx, taskID)
+	if err != nil {
+		return nil, err
+	}
+	return r, c.sch.UpdateTask(t, m)
```

Comment:
I see that calling UpdateTask cancels in-flight work:

https://github.com/influxdata/influxdb/blob/ab2fd3ba2fee091c70c5988da784a7497c1c4942/task/backend/scheduler.go#L335-L339

IIRC a canceled run won't be restarted. What do you think about adding a new method on the `TickScheduler`, like `CheckManualRuns(taskID platform.ID)`? That would need to call `taskScheduler.Start` and make sure that the `taskScheduler.hasQueue` is set to true so it knows to call `DesiredState.CreateNextRun`.

Reply:
Seems reasonable. Do you think it should be added to the Scheduler interface or just the implementation of the TickScheduler? Im leaning toward the tick scheduler, only because this manual run problem is a consequence of the tick scheduler's caching of the run data. Thoughts?

Reply:
or maybe the update task shouldnt cancel work in progress. It seems like any work in progress should proceed and any new work should get the update?

Reply:
I vaguely recall that we consciously decided that Update should cancel in progress work, because we can't make assumptions about what changed during an update. 

Without playing with the code, it's hard for me to say whether just updating TickScheduler will work out cleanly as opposed to updating the Scheduler interface too. I don't know if there's enterprise code which refers to the Scheduler interface, and which would need to know about this new method.

But if it works out fine just updating TickScheduler, I'm okay with that approach.

Reply:
After Looking back at the reason why we replaced the task scheduler (https://github.com/influxdata/platform/pull/1058). Im now sure we wrote it to replace the task scheduler as a quick solution not because of any specific intent. I think It would be quite simple to update the `UpdateTask` to not replace but instead update.

Reply:
Okay, fine by me.

Reply:
I prefer this behavior to canceling.